src/endpoints/unit/sell-products/selling-controller.ts

```diff
@@ -0,0 +1,114 @@
+import {Body, Controller, Post} from '@nestjs/common';
+import {SaleService} from 'src/endpoints/sale/sale.service';
+import {Sale, SalePaymentMethod, SaleProduct, SaleRequest} from 'src/models';
+import {ClsService} from 'nestjs-cls';
+import {v4 as uuidv4} from 'uuid';
+import {SaleProductService} from 'src/endpoints/sale-product/sale-product.service';
+import {ProductService} from 'src/endpoints/product/product.service';
+import {SalePaymentMethodService} from 'src/endpoints/sale-payment-method/sale-payment-method.service';
+import {error} from 'console';
+import {PaymentMethodService} from 'src/endpoints/payment-method/payment-method.service';
+
+@Controller('api/selling')
+export class SellingController {
+  constructor(
+    private readonly saleService: SaleService,
+    private readonly cls: ClsService,
+    private readonly saleProductService: SaleProductService,
+    private readonly productService: ProductService,
+    private readonly salePaymentMethodService: SalePaymentMethodService,
+    private readonly paymentMethodService: PaymentMethodService,
+  ) {}
+
+  @Post('sellProducts')
+  async sellProducts(@Body() saleRequest: SaleRequest) {
+    const prices = saleRequest.products.map(
+      x => x.sellingPrice * x.productCount,
+    );
+    const total = prices.reduce(
+      (accumulator, currentValue) => accumulator + currentValue,
+    );
+    const products = await this.productService.getAll();
+    const sale = new Sale();
+    const paymentMethods = await this.paymentMethodService.getAll();
+
+    if (
+      total ==
+      saleRequest.paymentMethods
+        .map(x => x.amount)
+        .reduce((accumulator, currentValue) => accumulator + currentValue)
+    ) {
+      throw new error('selling-product.payments-non-valid');
+    }
+
+    if (total <= 0) {
+      throw new error('selling-controller.total-not-valid');
+    }
+    
+    if (
+      saleRequest.products
+        .map(x => x.productCount)
+        .reduce((accumulator, currentValue) => accumulator + currentValue) >= 1
```

Comment:
Extract long code into a new variable to have a clean/readable code.
The same rule applies to the other code.